tests/components/zha/test_config_flow.py

```diff
@@ -1,125 +1,192 @@
 """Tests for ZHA config flow."""
-from unittest import mock
+
+import os
+
+from asynctest import mock
+import serial.tools.list_ports
+import zigpy.config
 
 from homeassistant.components.zha import config_flow
-from homeassistant.components.zha.core.const import CONTROLLER, DOMAIN, ZHA_GW_RADIO
-import homeassistant.components.zha.core.registries
+from homeassistant.components.zha.core.const import CONF_RADIO_TYPE, CONTROLLER, DOMAIN
 
 import tests.async_mock
 from tests.common import MockConfigEntry
 
 
+def com_port():
+    """Mock of a serial port."""
+    port = serial.tools.list_ports_common.ListPortInfo()
+    port.serial_number = "1234"
+    port.manufacturer = "Virtual serial port"
+    port.device = "/dev/ttyUSB1234"
+    port.description = "Some serial port"
+
+    return port
+
+
+@mock.patch(
+    "serial.tools.list_ports.comports", mock.MagicMock(return_value=[com_port()])
+)
 async def test_user_flow(hass):
-    """Test that config flow works."""
+    """Test user flow."""
     flow = config_flow.ZhaFlowHandler()
     flow.hass = hass
 
-    with tests.async_mock.patch(
-        "homeassistant.components.zha.config_flow.check_zigpy_connection",
-        return_value=False,
+    port = com_port()
+    port_select = f"{port}, s/n: {port.serial_number} - {port.manufacturer}"
+
+    with mock.patch.object(
+        flow, "detect_radios", return_value=mock.sentinel.data,
     ):
         result = await flow.async_step_user(
```

Comment:
Please use the core interface `hass.config_entries.flow.async_init`.

Reply:
is there a test testing a multi-step config entry I could use as a reference?

Reply:
Used direct_tv test as reference.